1000genomes_phase3/README.Rmd

```diff
@@ -0,0 +1,116 @@
+<!-- WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. -->
+<!-- See the License for the specific language governing permissions and -->
+<!-- limitations under the License. -->
+
+1,000 Genomes Phase 3 Variants
+==============================
+
+### Additional Resources
+* [Schema](https://bigquery.cloud.google.com/table/genomics-public-data:1000_genomes_phase_3.variants_20150220_release?pli=1)
+* [Data Provenance](http://googlegenomics.readthedocs.io/en/latest/use_cases/discover_public_data/1000_genomes.html)
+* [Index of variant analyses](./sql)
+
+### Metrics
+
+In the following plots we display metrics computed on both 1,000 Genomes phase 1 and phase 3 variants.
+
+```{r echo=FALSE, eval=FALSE}
+######################[ CHANGE ME ]##################################
+# This codelab assumes that the current working directory is where the Rmd file resides.
+setwd("/YOUR/PATH/TO/bigquery-examples/1000genomes_phase3")
+
+# Set the Google Cloud Platform project id under which these queries will run.
+project <- "YOUR-PROJECT-ID"
+#####################################################################
+```
+
+```{r init, echo=FALSE, message=FALSE, warning=FALSE, comment=NA}
+library(bigrquery)
+library(ggplot2)
+library(dplyr)
+library(xtable)
+```
+
+```{r echo=FALSE, message=FALSE, warning=FALSE}
+sample_info <- read.csv("http://storage.googleapis.com/genomics-public-data/1000-genomes/other/sample_info/sample_info.csv")
+phase3 <- query_exec(
+  "SELECT * FROM [google.com:biggene:1000genomes_analysis_results.phase3_metrics]",
+  project)
+phase1 <- query_exec(
+  "SELECT * FROM [google.com:biggene:1000genomes_analysis_results.phase1_metrics]",
+  project)
+results <- inner_join(sample_info,
+                      rbind_list(mutate(phase3, dataset = "phase3"),
+                                 mutate(phase1, dataset = "phase1")),
+                      by=c("Sample" = "call_call_set_name"))
+```
+
+```{r titv_metrics, echo=FALSE, message=FALSE, warning=FALSE, comment=NA, fig.align="center", fig.width=12, fig.height=8}
+ggplot(results, aes(dataset, ti_tv_ratio)) +
+  geom_boxplot(aes(colour = Super_Population)) +
+  ylab("Het/Hom Ratio") +
+  xlab("Dataset") +
+  ggtitle("Transition/Transversion SNP Ratio") 
+
+ggplot(results, aes(dataset, ti_tv_ratio)) +
+  geom_boxplot(aes(colour = Gender)) +
+  ylab("Het/Hom Ratio") +
+  xlab("Dataset") +
+  ggtitle("Transition/Transversion SNP Ratio") 
+```
+
+```{r hethom_metrics, echo=FALSE, message=FALSE, warning=FALSE, comment=NA, fig.align="center", fig.width=12, fig.height=8}
+ggplot(results, aes(dataset, het_hom_ratio)) +
+  geom_boxplot(aes(colour = Super_Population)) +
+  ylab("Ti/Tv Ratio") +
+  xlab("Dataset") +
+  ggtitle("Heterozygous/Homozygous Variant Ratio") 
+
+ggplot(results, aes(dataset, het_hom_ratio)) +
+  geom_boxplot(aes(colour = Gender)) +
+  ylab("Ti/Tv Ratio") +
+  xlab("Dataset") +
+  ggtitle("Heterozygous/Homozygous Variant Ratio") 
+```
```

Comment:
Hi Nicole,

Something doesn't look quite right with some of the plots.  So if in your query you have the following definitions on [lines 11-12](https://github.com/deflaux/bigquery-examples/blob/b180ed78a5b7127d985961efc028e746fbfed7f5/1000genomes_phase3/sql/qc-metrics.sql#L11-12):

```
transitions_count / transversions_count AS ti_tv_ratio,
(het_RA_count + het_AA_count) / (hom_RR_count + hom_AA_count) AS het_hom_ratio,
```

Why are the y-axes labels then not matching in your ggplots's y-axis input variables for the following four plots:

```R
ggplot(results, aes(dataset, ti_tv_ratio)) +
  geom_boxplot(aes(colour = Super_Population)) +
  ylab("Het/Hom Ratio") +
  xlab("Dataset") +
  ggtitle("Transition/Transversion SNP Ratio") 

ggplot(results, aes(dataset, ti_tv_ratio)) +
  geom_boxplot(aes(colour = Gender)) +
  ylab("Het/Hom Ratio") +
  xlab("Dataset") +
  ggtitle("Transition/Transversion SNP Ratio") 

ggplot(results, aes(dataset, het_hom_ratio)) +
  geom_boxplot(aes(colour = Super_Population)) +
  ylab("Ti/Tv Ratio") +
  xlab("Dataset") +
  ggtitle("Heterozygous/Homozygous Variant Ratio") 

ggplot(results, aes(dataset, het_hom_ratio)) +
  geom_boxplot(aes(colour = Gender)) +
  ylab("Ti/Tv Ratio") +
  xlab("Dataset") +
  ggtitle("Heterozygous/Homozygous Variant Ratio") 

```

  ~p